analyzers/nodejs/integration_test.go

```diff
@@ -0,0 +1,210 @@
+	// faker has no deps
+	// assertProjectFixtureExists(t, "fakerjs")
+	assertProjectFixtureExists(t, "fastify")
+	assertProjectFixtureExists(t, "nest")
+	assertProjectFixtureExists(t, "ohm")
+	assertProjectFixtureExists(t, "express")
+	assertProjectFixtureExists(t, "standard")
+	assertProjectFixtureExists(t, "sodium-encryption")
+	assertProjectFixtureExists(t, "request")
+}
+
+func TestAnalysisOutput(t *testing.T) {
+	t.Parallel()
+	for _, proj := range projects {
+		t.Run(proj.Name, func(t *testing.T) {
+			module := module.Module{
+				Dir:         filepath.Join(nodeAnalyzerFixtureDir, proj.Name),
+				Type:        pkg.NodeJS,
+				Name:        proj.Name,
+				Options:     map[string]interface{}{"allow-npm-err": true},
+				BuildTarget: filepath.Join(nodeAnalyzerFixtureDir, proj.Name, "package.json"),
+			}
+
+			analyzer, err := analyzers.New(module)
+			assert.NoError(t, err)
+
+			deps, err := analyzer.Analyze()
+			assert.NoError(t, err)
+			// faker has no deps
+			if proj.Name == "fakerjs" {
+				assert.Empty(t, deps.Direct)
+				assert.Empty(t, deps.Transitive)
+			} else {
+				assert.NotEmpty(t, deps.Direct)
+				assert.NotEmpty(t, deps.Transitive)
+			}
+		})
+	}
+}
+
+func assertProjectFixtureExists(t *testing.T, name string) {
+	exists, err := files.ExistsFolder(nodeAnalyzerFixtureDir, name)
+	assert.NoError(t, err)
+	assert.True(t, exists, name+" was not properly cloned")
+
+	exists, err = files.ExistsFolder(nodeAnalyzerFixtureDir, name, "node_modules")
+	assert.NoError(t, err)
+	assert.True(t, exists, name+" did not have its node modules installed")
+}
+
+func initializeProjects(testDir string) error {
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(projects))
+
+	for _, project := range projects {
+		go func(proj testUtils.ProjectFixture) {
+			defer waitGroup.Done()
+
+			projectDir := filepath.Join(testDir, proj.Name)
+
+			_, errOut, err := exec.Run(exec.Cmd{
+				Name:    "npm",
+				Argv:    []string{"install", "--production"},
+				Dir:     projectDir,
+				WithEnv: proj.Env,
+				Command: "npm",
+			})
+			if err != nil {
+				println(errOut)
+				println("failed to run npm install on " + proj.Name)
+			}
+
+			// save time on local
+			ymlAlreadyExists, err := files.Exists(filepath.Join(projectDir, ".fossa.yml"))
+			if err != nil {
+				panic(err)
+			}
+			if ymlAlreadyExists {
+				return
+			}
+
+			// any key will work to prevent the "NEED KEY" error message
+			err = testUtils.FossaInit(projectDir)
+			if err != nil {
+				println("failed to run fossa init on " + proj.Name)
+				println(err.Error())
+				panic(err)
+			}
+		}(project)
+	}
+
+	waitGroup.Wait()
+
+	return nil
+}
+
+var projects = []testUtils.ProjectFixture{
+	testUtils.ProjectFixture{
+		Name:   "puppeteer",
+		URL:    "https://github.com/GoogleChrome/puppeteer",
+		Commit: "b97bddf8e5750d20c6ba82392eebe2a3fd2dd218",
+		Env: map[string]string{
+			"PUPPETEER_SKIP_CHROMIUM_DOWNLOAD": "1",
+		},
+	},
+	testUtils.ProjectFixture{
```

Comment:
Is it preferable to pull these fixtures from NPM or GitHub? I can see arguments for both, but I'm wondering what your thoughts are.

Reply:
I think we should pull this from git. We can version them ourselves and it gives us control. Pulling from npm means that we'll always be reliant on another toolchain we don't control

Additionally, pulling from git allows for a consistent repo for all fixtures, regardless of ecosystem